test/Compressors/sampling.jl

```diff
@@ -0,0 +1,501 @@
+module Sampling_compressor
+using Test, RLinearAlgebra, Random
+using StatsBase: ProbabilityWeights, sample
+import LinearAlgebra: mul!, Adjoint
+using ..FieldTest
+using ..ApproxTol
+
+Random.seed!(2131)
+@testset "Sampling" begin
+    @testset "Sampling: Compressor" begin
+        # Verify Supertype
+        @test supertype(Sampling) == Compressor
+
+        # Verify fields and types
+        @test fieldnames(Sampling) == (:cardinality, :compression_dim, :distribution)
+        @test fieldtypes(Sampling) == (Cardinality, Int64, Distribution)
+
+        # Verify the Internal Constructor
+        let cardinality = Left(), compression_dim = 0, distribution = Uniform()
+            @test_throws ArgumentError(
+                "Field `compression_dim` must be positive."
+            ) Sampling(
+                cardinality, compression_dim, distribution
+            )
+        end
+
+        let cardinality = Undef(), compression_dim = 2, distribution = Uniform()
+            @test_throws ArgumentError(
+                "`cardinality` must be specified as `Left()` or `Right()`.\
+                `Undef()` is not allowed in `CountSketch` structure."
+            ) Sampling(
+                cardinality, compression_dim, distribution
+            )
+        end
+
+        # Default values
+        let sc_default = Sampling()
+            @test sc_default.cardinality == Left()
+            @test sc_default.compression_dim == 2
+            @test sc_default.distribution isa Uniform
+            # Check defaults of the default Uniform distribution instance
+            @test sc_default.distribution.cardinality == Undef()
+            @test sc_default.distribution.replace == false
+        end
+
+        # Specific cardinality
+        let card_val = Right()
+            dist_instance = Uniform(cardinality=card_val) 
```

Comment:
the `Uniform` constructor is not what is being tested here, so it should be a parameter of the `let` block.